packages/core/src/dispatcher.ts

```diff
@@ -64,12 +70,26 @@ export default class Dispatcher extends EventEmitter {
    */
   async register (document: Document): Promise<void> {
     this._documents[document.id.toString()] = document
-    // request tip
-    const payload = { typ: MsgType.REQUEST, id: document.id.toString(), doctype: document.doctype.doctype }
+
+    // Build a QUERY message to send to the pub/sub topic to request the latest tip for this document
+    const payload = await this._buildQueryMessage(document)
+
+    // Store the query id so we'll process the corresponding RESPONSE message when it comes in
+    this._outstandingQueryIds[payload.id] = document.id.baseID
+
     this._ipfs.pubsub.publish(this.topic, JSON.stringify(payload))
     this._log({ peer: this._peerId, event: 'published', topic: this.topic, message: payload })
   }
 
+  async _buildQueryMessage(document: Document): Promise<Record<string, any>> {
+    const message = { typ: MsgType.QUERY, doc: document.id.baseID.toString() }
+
+    // Add 'id' to message.  'id' is the multihash of the rest of the query object canonicalized
+    // using dag-cbor
+    const id: Uint8Array = dagCBOR.util.serialize(message)
```

Comment:
Please compute a multihash of the serialized message. Now the `id` is the entire message serialized as dag-cbor.

Reply:
Okay new version is up.  Is this what you had in mind @oed?